src/renderer/services/file-management-system/index.ts

```diff
@@ -398,85 +397,134 @@ export default class FileManagementSystem {
    * already done to upload the file.
    */
   private async resume(
-    upload: UploadJob,
+    fuaUpload: UploadJob,
     fssStatus: UploadStatusResponse,
     onProgress: (uploadId: string, progress: UploadProgressInfo) => void
   ): Promise<void> {
-    const { fssUploadId, lastModifiedInMS, files } = upload.serviceFields;
-    const { mtime: fileLastModified } =
-      await fs.promises.stat(files[0].file.originalPath);
+    const { localNasShortcut, lastModifiedInMS, files } = fuaUpload.serviceFields;
+    const { mtime: fileLastModified } = await fs.promises.stat(files[0].file.originalPath);
     const fileLastModifiedInMs = fileLastModified.getTime();
-
     if (lastModifiedInMS !== fileLastModifiedInMs) {
       throw new Error("File has been modified since last upload attempt");
     }
-
-    // Skip trying to resume if there is no FSS upload to check
-    if (fssUploadId) {
-      // Retrive job FSS uses to track its upload portion
-      let fssUpload;
-      try {
-        fssUpload = (await this.jss.getJob(fssUploadId)) as FSSUpload;
-      } catch (error) {
-        // Because FSS uses a queue to interact with JSS this job
-        // may not exist even though the upload is in progress
-        // in which case the app can just wait it out
-        console.error("Failed to get FSS job")
-        return;
-      }
-
-      if (fssStatus.status === UploadStatus.COMPLETE) {
+    switch (fssStatus.status) {
+      case UploadStatus.WORKING:
+        // Update status to reflect the resume going smoothly
+        await this.jss.updateJob(fuaUpload.jobId, {
+          status: JSSJobStatus.RETRYING,
+        });
+        if (localNasShortcut) {
+          // For localNasShortcut uploads, the way to reume an in progress upload is to call /register on it again. 
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          const [registration, _s, fileSize] = await this.register(fuaUpload);
+          this.updateProgress(registration.uploadId, fileSize, (progress: UploadProgressInfo) => onProgress(registration.uploadId, progress))
+        } else {
+          await this.resumeUploadInChunks(fuaUpload, fssStatus, onProgress);
+        }
+        break;
+      case UploadStatus.RETRY:
+        if (localNasShortcut) {
+          await this.fss.retryFinalizeLocalNasShortcut(fssStatus.uploadId);
```

Comment:
I understand that if a chunk needs to be retried the whole job will present as `WORKING` still in the chunked scenario, but what happens in the local NAS shortcut scenario when something goes awry during the copy? Does it just get marked as `INACTIVE`?

Reply:
`RETRY` is for when a post-upload task fails (and exceeds retries, like if MMS is down).  The localNasShortcut 'copy' is more of a read (by FSS2), and then upload chunks using the existing infrastructure.  If something goes awry, there will just be missing chunks (just like with the chunked upload).